src/components/Menu/MenuItem/MenuItem.tsx

```diff
@@ -14,46 +22,89 @@ import usePopover from "../../../hooks/usePopover";
 import { backwardCompatibilityForProperties } from "../../../helpers/backwardCompatibilityForProperties";
 import useMenuItemMouseEvents from "./hooks/useMenuItemMouseEvents";
 import useMenuItemKeyboardEvents from "./hooks/useMenuItemKeyboardEvents";
+import { SubIcon, VibeComponent, VibeComponentProps } from "../../../types";
+import { IconType } from "../../Icon/IconConstants";
+import { CloseMenuOption } from "../Menu/MenuConstants";
 import "./MenuItem.scss";
 
 const TAB_INDEX_FOCUS_WITH_JS_ONLY = -1;
 
-const MenuItem = forwardRef(
+export interface MenuItemProps extends VibeComponentProps {
+  title?: string;
+  label?: string;
+  icon?: SubIcon;
+  iconType?: IconType;
+  iconBackgroundColor?: string;
+  disabled?: boolean;
+  disableReason?: string;
+  selected?: boolean;
+  onClick?: (event: React.MouseEvent | React.KeyboardEvent) => void;
+  activeItemIndex?: number;
+  setActiveItemIndex?: (index: number) => void;
+  index?: number;
+  key?: string;
+  isParentMenuVisible?: boolean;
+  resetOpenSubMenuIndex?: () => void;
+  hasOpenSubMenu?: boolean;
+  setSubMenuIsOpenByIndex?: (index: number, isOpen: boolean) => void;
+  useDocumentEventListeners?: boolean;
+  tooltipContent?: string;
+  tooltipPosition?: typeof MenuItem.tooltipPositions[keyof typeof MenuItem.tooltipPositions];
+  tooltipShowDelay?: number;
+  onMouseLeave?: (event: React.MouseEvent) => void;
+  onMouseEnter?: (event: React.MouseEvent) => void;
+  /** Backward compatibility for props naming **/
+  classname?: string;
+  menuId?: string;
+  isInitialSelectedState?: boolean;
+  shouldScrollMenu?: boolean;
+  closeMenu?: (option: CloseMenuOption) => void;
+  menuRef?: React.RefObject<HTMLElement>;
+  children?: ReactElement | ReactElement[];
+}
+
+const MenuItem: VibeComponent<MenuItemProps> & {
+  iconType?: typeof Icon.type;
+  tooltipPositions?: typeof DialogPosition;
+  isSelectable?: boolean;
+  isMenuChild?: boolean;
```

Comment:
Why not include these in `MenuItemProps`?

Reply:
static props